pkg/trace/agent/agent_test.go

```diff
@@ -404,6 +404,68 @@ func TestProcess(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("filtering", func(t *testing.T) {
+		for _, tt := range []struct {
+			trace pb.Trace
+			f     func(t *testing.T, spans []*pb.Span)
+		}{
+			{
+				trace: pb.Trace{
+					{
+						TraceID: 1,
+						SpanID:  1,
+						Meta:    map[string]string{"db": "sql", "env": "testing"},
+					},
+					{
+						TraceID:  1,
+						SpanID:   2,
+						ParentID: 1,
+						Meta:     map[string]string{"env": "prod", "db": "mongodb"},
+					},
+				},
+				f: func(t *testing.T, spans []*pb.Span) {
+					assert.Equal(t, int(2), len(spans))
+				},
+			},
+			{
+				trace: pb.Trace{
+					{
+						TraceID: 1,
+						SpanID:  3,
+						Meta:    map[string]string{"env": "fake", "db": "bad-value"},
+					},
+					{
+						TraceID:  1,
+						SpanID:   4,
+						ParentID: 1,
+						Meta:     map[string]string{"env": "prod", "db": "mysql"},
+					},
+				},
+				f: func(t *testing.T, spans []*pb.Span) {
+					assert.Equal(t, int(2), len(spans))
```

Comment:
Well, what exactly is being tested here? Except the length of the original trace passed to `Process`? 🙂 The way to test the `Process` function is to receive on the `a.TraceWriter.In` channel, which is the channel where all traces sent to Datadog are sent too. That channel should receive all traces, _except_ the filtered ones. That's how the test needs to be written. In order to make sure nothing else in the `Process` function rejects the trace, make sure to add the `Metrics` tag `"_sampling_priority_v1": 2`

Reply:
I'll remove the test